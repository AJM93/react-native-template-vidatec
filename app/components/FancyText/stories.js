/*
 * app/components/FancyText/stories.js
 */

import React from 'react';
import { View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import FancyText from '.';

storiesOf('FancyText', module)
  .addDecorator(getStory => <View>{getStory()}</View>)
  .add('no props', () => (
    <FancyText />
  ));