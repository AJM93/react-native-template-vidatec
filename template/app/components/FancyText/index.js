/*
 * app/components/FancyText/index.js
 */

// import react and react-native elements
import React, { Component } from 'react';
import { Text } from 'react-native';

// import screens styles
import styles from './styles';

class FancyText extends Component {
  /**
   * Construct component class
   * @param {object} props
   */
  constructor (props: {}) {
    super(props);

    this.state = {
      // store component state here
    }
  }

  /**
   * Render component
   */
  render () {
    return (
        <Text style={styles.text}>{this.props.children}</Text>
    );
  }
}

export default FancyText;
