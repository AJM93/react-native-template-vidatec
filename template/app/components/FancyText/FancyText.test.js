/*
 * app/components/FancyText/FancyText.test.js
 */

import React from 'react';
import renderer from 'react-test-renderer';

import FancyText from '.';

it('renders correctly with no props', () => {
  const tree = renderer.create(<FancyText />).toJSON();
  expect(tree).toMatchSnapshot();
});
