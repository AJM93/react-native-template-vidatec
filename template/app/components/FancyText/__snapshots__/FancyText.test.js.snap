// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no props 1`] = `
<Text
  style={
    Object {
      "color": "#FF0000",
      "fontSize": 20,
      "textAlign": "center",
    }
  }
/>
`;
